### PR TITLE
Add like/dislike buttons in vscode status bar

### LINKED ISSUE #98: Like/Dislike buttons
Would be awesome to have the like/dislike buttons directly in vscode.

I'll see if I have the time to look at it this weekend, might send a PR.

### PR DESCRIPTION
**Pull request by cleferman**
Fixes #98

Solid thumbs-up/down icons are not present in vscode product icons. Decided to go with MUI font since it's the same font used by the desktop player.
Defined both outlined and solid thumbs for consistency.

Screenshot:
image

### CODE REVIEW

File: src/youtubeMusic.ts [hunk 1]
```diff
@@ -114,6 +116,20 @@ export default class YouTubeMusic {
       },
       { id: "skip", title: "Next Song", text: "$(chevron-right)" },
       { id: "cycleRepeat", title: "Not Repeating", text: "$(sync)" },
+      {
+        id: "thumbsdown",
+        title: "Thumbs Down",
+        text: "$(mui-thumbs-down)",
+        dynamicText: (isThumbsDown: boolean) =>
+        isThumbsDown ? "$(mui-thumbs-down-solid)" : "$(mui-thumbs-down)",
```
Comment (sedders123) [inline src/youtubeMusic.ts:124]: Could we stick to consistent whitespace here (matching `thumbsup` below)
```suggestion
        dynamicText: (isThumbsDown: boolean) =>
          isThumbsDown ? "$(mui-thumbs-down-solid)" : "$(mui-thumbs-down)",
```

File: package.json [hunk 2]
```diff
@@ -71,7 +71,44 @@
...
       }
-    ]
+    ],
+    "productIconThemes": [
+      {
+        "id": "material-ui-font-icons",
+        "label": "Material UI font Icons",
+        "path": "./productIcons/material-ui-font-icons.json"
+      }
+    ],
```
Comment (sedders123) [inline package.json:81]: I'm not sure if we need this. [The docs](https://code.visualstudio.com/api/extension-guides/product-icon-theme) suggest this is for specifying that we expose a theme which I don't think we do. Also `./productIcons/material-ui-font-icons.json` is not added by this PR and didn't already exist.
```suggestion
```

**Review by sedders123 — COMMENTED** [review]
Firstly thanks for submitting this feature! Really cool to see that this extension is useful to others. Unfortunately there are a couple of sticking points before I feel happy merging this in. More than happy to discuss any of the below in more detail!

## Bundle Size

In its current state this PR increases the size of the package from 98.7 KB to 225 KB which is quite the bump. Pretty much all this come from including the material-ui font.

There are a few ways we could potentially reduce the size:
### Use icons that are already in the vscode icon set
This isn't great as as you say there are not thumbs up or thumbs down buttons

We could use something like up and down arrows but this could lose the boldness for showing state (not looked at all the arrows and their thicknesses). 

Though losing boldness could be fine if maybe it became one button with three states (liked - up arrow, dislike - down arrow, neither - not sure of icon)? Clicking this button could then cycle the state but ads complexity to our implementation with the YTMusic API. Not sure how I feel about the one button approach but its an option.

### Change the font file so that it only contains the 4 icons we need
I believe there are ways to mangle the font file so that it only contains the characters we need. This would mean that if we wanted to use other icons in the future we'd have more work, but it would be pretty effective at cutting the size

### Don't have buttons
Obviously not a great option, but we could not show anything in the status bar and instead just add like/dislike commands that can be mapped to keyboard shortcuts

## Licensing

I'm not sure what license the `material-ui-font.woff2` included in this PR is distributed under. Are we allowed to bundle it into our own package, do we need to add disclaimers anywhere. If we need a disclaimer in the readme that's likely fine but if we need a disclaimer that could be accessed from within the extension while its running that's a fair bit more work.

## Next Steps
Keen to hear what you think the best way forward is. Personally I think I'm leaning towards the font mangling route if the license stuff is all good, otherwise in built icons in some form.

File: package.json (hunk 2 above)
Comment (cleferman) [reply package.json:81]: good catch, it's a remnant from before defining the icons. I missed removing it before the final commit

**Comment by cleferman** [comment]
Thanks for your feedback!

The license for the font seems to be `SIL Open Font License (OFL)`. I just briefly read it and we seem to be in the clear. 
I also found this cool [FAQ ](https://scripts.sil.org/cms/scripts/page.php?item_id=OFL-FAQ_web#1b11b702) which answers our questions. Question 2.2 and 2.6 are relevant for us.

There doesn't seem to be anything about adding a disclaimer.

I'm not sure when I get the time to look at modifying the font. Hoping sooner than later :)

**Comment by cleferman** [comment]
@sedders123 managed to subset the font and only include the 4 icons.

**Review by sedders123 — COMMENTED** [review]
Awesome, thanks so much for that! The package is now at 100 KB which is a huge improvement :)

Tested this locally and all seems good. One final tweak to the version number and then I think we're good to merge!

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "vscode-ytmusic",
   "displayName": "vscode-ytmusic",
   "description": "Connecting VS Code with the YTMDesktop Player",
-  "version": "1.2.2",
+  "version": "1.2.3",
```
Comment (sedders123) [inline package.json:5]: Was thinking that this feature would probably be warrant a minor bump so it'd be 1.3.0. Ideally version bumps should be in their own commit, feel free to either do as part of this PR, or switch back to 1.2.2 and I can add a commit once merged :)

Comment (cleferman) [reply package.json]: alright, updated it as part of this PR

**Review by sedders123 — APPROVED** [review]

**Comment by sedders123** [comment]
Thanks so much for your work on this one, really appreciate it! Will get a release out with this in later this evening :)

**Comment by cleferman** [comment]
Thank you for your work on the desktop app and the extension. I didn't do much to be honest. Happy to help! :)

**Pull request merged**